spring-core/src/main/java/org/springframework/util/StreamUtils.java

```diff
@@ -238,6 +239,24 @@ public static OutputStream nonClosing(OutputStream out) {
 		return new NonClosingOutputStream(out);
 	}
 
+	/**
+	 * More effective equivalent of {@code new String(baos.toByteArray(), charset)}
+	 * As far as at invocation point {@code charset} is already available,
+	 * no exception is expected to be thrown.
+	 *
+	 * @param baos    {@link ByteArrayOutputStream} to be flushed into String
+	 * @param charset applicable {@link Charset}
+	 * @return String represenation of bytes stored in {@code baos}
+	 */
+	public static String baosToString(ByteArrayOutputStream baos, Charset charset) {
```

Comment:
I think it would be better to simply overload the existing `copyToString(...)` method rather than introduce a new `baosToString(...)` variant with a different name but the same purpose.

In any case, I'll merge the work as-is and perform the rename afterwards.

Reply:
Good point, done.